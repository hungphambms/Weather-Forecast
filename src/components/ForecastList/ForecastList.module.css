.list {
    padding: 0 24px;
    max-width: 480px;
    margin: 0 auto;
}