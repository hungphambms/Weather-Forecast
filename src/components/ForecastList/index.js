// import React from "react";
import PropTypes from "prop-types";

import ForecastItem from "../ForecastItem";

import styles from "./ForecastList.module.css";

function ForecastList({ items }) {
  return (
    <div className={styles.list}>
      {items.map((item) => (
        <ForecastItem key={item.dt} item={item} />
      ))}
    </div>
  );
}

ForecastList.defaultProps = {
  items: [],
};

ForecastList.propTypes = {
  items: PropTypes.array,
};

export default ForecastList;
