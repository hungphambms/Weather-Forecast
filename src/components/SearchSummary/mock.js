export const MOCK_DATA = {
  id: 1566083,
  name: "Ho Chi Minh City",

  lat: 10.75,
  lon: 106.6667,
  timezone: "Asia/Ho_Chi_Minh",
  timezone_offset: 25200,
  current: {
    dt: 1615014591,
    sunrise: 1614985544,
    sunset: 1615028631,
    temp: 307.15,
    feels_like: 307.65,
    pressure: 1011,
    humidity: 36,
    dew_point: 289.98,
    uvi: 9.28,
    clouds: 40,
    visibility: 10000,
    wind_speed: 2.57,
    wind_deg: 40,
    weather: [
      {
        id: 802,
        main: "Clouds",
        description: "scattered clouds",
        icon: "03d",
      },
    ],
  }
};
