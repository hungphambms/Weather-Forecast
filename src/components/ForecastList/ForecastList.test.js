import { render, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';

import {MOCK_DATA} from "./mock";

import ForecastList from './index';

describe('ForecastList rendering', () => {
  it('renders ForecastList without crashing', () => {
    const render = renderer.create(<ForecastList items={MOCK_DATA.daily} />);
    const tree = render.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correct items', async() => {
    const component = await waitFor(() => render(<ForecastList items={MOCK_DATA.daily} />));
    const listItems = component.getAllByLabelText('item');

    expect(listItems).toHaveLength(MOCK_DATA.daily.length);
  });

  it('renders null when item is null', () => {
    const component = render(<ForecastList />);
    expect(component.textContent).toBe(undefined);
  });
});
