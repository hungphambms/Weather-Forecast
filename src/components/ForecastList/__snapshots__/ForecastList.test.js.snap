// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastList rendering renders ForecastList without crashing 1`] = `
<div
  className="list"
>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-06T05:00:00.000Z"}
      >
        Saturday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-06T05:00:00.000Z"}
      >
        Mar 06, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      22
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      36
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/01d.png"
      />
      36
    </label>
  </div>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-07T05:00:00.000Z"}
      >
        Sunday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-07T05:00:00.000Z"}
      >
        Mar 07, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      23
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      38
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/02d.png"
      />
      36
    </label>
  </div>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-08T05:00:00.000Z"}
      >
        Monday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-08T05:00:00.000Z"}
      >
        Mar 08, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      23
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      38
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/01d.png"
      />
      35
    </label>
  </div>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-09T05:00:00.000Z"}
      >
        Tuesday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-09T05:00:00.000Z"}
      >
        Mar 09, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      24
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      36
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/03d.png"
      />
      36
    </label>
  </div>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-10T05:00:00.000Z"}
      >
        Wednesday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-10T05:00:00.000Z"}
      >
        Mar 10, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      24
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      37
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/01d.png"
      />
      37
    </label>
  </div>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-11T05:00:00.000Z"}
      >
        Thursday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-11T05:00:00.000Z"}
      >
        Mar 11, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      24
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      37
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/03d.png"
      />
      37
    </label>
  </div>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-12T05:00:00.000Z"}
      >
        Friday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-12T05:00:00.000Z"}
      >
        Mar 12, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      25
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      37
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/03d.png"
      />
      37
    </label>
  </div>
  <div
    aria-label="item"
    className="item"
  >
    <h3>
      <time
        dateTime={"2021-03-13T05:00:00.000Z"}
      >
        Saturday
      </time>
    </h3>
    <p>
      <time
        dateTime={"2021-03-13T05:00:00.000Z"}
      >
        Mar 13, 2021
      </time>
    </p>
    <p>
      <img
        alt="min icon"
        src="min.png"
      />
      <strong>
        Min
      </strong>
       
      24
      <span
        className="divider"
      />
      <img
        alt="max icon"
        src="max.png"
      />
      <strong>
        Max
      </strong>
       
      36
    </p>
    <label>
      <img
        alt="Temp Icon"
        src="http://openweathermap.org/img/w/03d.png"
      />
      36
    </label>
  </div>
</div>
`;
